zproject/local_settings_template.py

```diff
@@ -148,39 +151,47 @@
 # The email gateway provides, for each stream, an email address that
 # you can send email to in order to have the email's content be posted
 # to that stream.  Emails received at the per-stream email address
-# will be converted into a Zulip message
-
+# will be converted into a Zulip message.
+#
 # There are two ways to make use of local email mirroring:
 #  1. Local delivery: A MTA runs locally and passes mail directly to Zulip
 #  2. Polling: Checks an IMAP inbox every minute for new messages.
-
-# A Puppet manifest for local delivery via Postfix is available in
-# puppet/zulip/manifests/postfix_localmail.pp. To use the manifest, add it to
-# puppet_classes in /etc/zulip/zulip.conf. This manifest assumes you'll receive
-# mail addressed to the hostname of your Zulip server.
-#
-# Users of other mail servers will need to configure it to pass mail to the
-# email mirror; see `python manage.py email-mirror --help` for details.
-
-# The email address pattern to use for auto-generated stream emails
-# The %s will be replaced with a unique token, and the resulting email
-# must be delivered to the EMAIL_GATEWAY_IMAP_FOLDER of the
-# EMAIL_GATEWAY_LOGIN account below, or piped in to the email-mirror management
-# command as indicated above.
 #
-# Example: zulip+%s@example.com
-EMAIL_GATEWAY_PATTERN = ""
-
-
-# The following options are relevant if you're using mail polling.
+# The local delivery configuration is preferred for production because
+# it doesn't have delay, while the polling mechanism is preferred for
+# development because it doesn't require an external DNS name.
 #
-# A sample cron job for mail polling is available at puppet/zulip/files/cron.d/email-mirror
+# The main email mirror setting is the email address pattern to use
+# for auto-generated stream emails.  The %s will be replaced with a
+# unique token, and the resulting email must be delivered to the
+# EMAIL_GATEWAY_IMAP_FOLDER of the EMAIL_GATEWAY_LOGIN account below,
+# or piped in to the email-mirror management command as indicated
+# above.
 #
+# E.g. %s@example.com (local delivery) or username+%s@example.com (polling)
```

Comment:
A clarification question: username+%s@example.com is basically required for a mailbox at username@example.com right? In order to get the mail delivered to username@example.com

Is %s@example.com the only option for local delivery or would username+%s@example.com also work for local delivery? I actually don't know the answer to this but I suspect the latter would work too. Personally, this is where some examples would have been useful, e.g. "If you have set up an IMAP mailbox at zulipadmin@mydomain.com, then using zulipadmin+%s@mydomain.com will generate stream email addresses that look like 'zulipadmin+streamidentifier+978699987@mydomain.com' and missed message reply email addresses that look like 'zulipadmin+089656647@mydomain.com'. If you set up local delivery, the same format would work, but you could also use "%s@mydomain.com" and the addresses would look like 'streamidentifier +978699987@mydomain.com' and '089656647@mydomain.com'"

I understand that that is somewhat implementation dependent, but this would have been much clearer to me :)